Plugins/Events/Events/QuickChatEvents.cpp

```diff
@@ -0,0 +1,61 @@
+#include "Events/QuickChatEvents.hpp"
+#include "API/CNWSPlayer.hpp"
+#include "API/CNWSMessage.hpp"
+#include "API/Functions.hpp"
+#include "API/Constants.hpp"
+#include "Events.hpp"
+#include "Utils.hpp"
+#include <cstring>
+
+
+namespace Events {
+
+using namespace NWNXLib;
+using namespace NWNXLib::API;
+using namespace NWNXLib::API::Constants;
+
+static NWNXLib::Hooking::FunctionHook* m_HandlePlayerToServerQuickChatMessageHook = nullptr;
+
+QuickChatEvents::QuickChatEvents(ViewPtr<Services::HooksProxy> hooker)
+{
+    hooker->RequestExclusiveHook<API::Functions::CNWSMessage__HandlePlayerToServerQuickChatMessage>(&HandlePlayerToServerQuickChatMessageHook);
+    m_HandlePlayerToServerQuickChatMessageHook = hooker->FindHookByAddress(API::Functions::CNWSMessage__HandlePlayerToServerQuickChatMessage);
+}
+
+template <typename T>
+static T PeekMessage(CNWSMessage *pMessage, int32_t offset)
```

Comment:
Think this can be moved to Utils?

Reply:
Yeah, it's used a whole bunch, I'll move it

Reply:
And moved